### ISSUE TITLE
No detection after training with Massachusetts Buildings Dataset

### ISSUE THREAD

**Issue by trecuu** [fuzailpalnak/building-footprint-segmentation#39]
Hello,

Im having a problem where I got no detection at all after training with Massachusetts Buildings Dataset.

Just to be sure, I should be using the chk_pth.pt as dict, right?

By the way, does the pretrained weight file should be working like "well"!? All I got  is something like this
![dst](https://user-images.githubusercontent.com/19274490/179808348-ce65769a-e19c-40db-a99d-73ad38f1ae60.jpg)
:

Thx for your work, and support!

**Comment by fuzailpalnak** [comment]
The reason for the pre-trained model to not yield results as expected could be that the model was trained on ariel imagery and this looks like a satellite imagery. Both are quite different visually as in ariel the building and its boundaries are evident while in the image you shared its does not as it is probably at zoom 15.

One thing I can suggest you do is to crop the image to a smaller dimension and do a image interpolation to get a image which has a similar resembles to Ariel imagery.


For training using pre-trained for Massachusetts dataset, I would suggest to go with ReFineNetLite, use augmentation and  apply regularisation as well.

Hope this helps

**Comment by fuzailpalnak** [comment]
Can you share the training configurations you are using ?

**Comment by trecuu** [comment]
Im using [Run with defined arguments.ipynb](https://github.com/fuzailpalnak/building-footprint-segmentation/blob/main/examples/Run%20with%20defined%20arguments.ipynb)

**Comment by fuzailpalnak** [comment]
Try out this few tricks and check if it helps 
1 - Make use of a lot of spatial Augmentation
2 - make use the augmentation which randomly crops the input during training and does resizing (i will have to check whats it called in albumentation lib)
3 - Apply regularization
4 - you can implement  this ( https://github.com/fuzailpalnak/building-footprint-segmentation/blob/e70778a1406a21472593faefc8af3cd85c8ee908/building_segmentation/ml/binary/models/albunet.py ) model if you are using the github code and not the Pip library 
5 - try with BCE loss

**Comment by fuzailpalnak** [comment]
@trecuu I trained a model with DlinkNet which is on [this branch](url) and the weights can be downloaded from [here](https://github.com/fuzailpalnak/building-footprint-segmentation/releases/download/alpha/DlinkNet.zip)
Hope this helps. 

![0_22678945_15](https://user-images.githubusercontent.com/24665570/182326740-6dca09d6-dbd5-4c1c-996a-08b642ace83e.png)
 
![0_22678945_15](https://user-images.githubusercontent.com/24665570/182326717-8637f8fd-75ac-4ba6-a116-04f663aceb50.png)

**Comment by trecuu** [comment]
I cant access "this branch" that you linked.
Is the link working for you?

Im new at ML, and I picked your REPO as a way to learn. So, Im kinda of lost... Sorry for the inconvenience.

Again, thx for the support you are giving.

**Comment by fuzailpalnak** [comment]
I have merged all the code to the main branch, so you can pull the latest changes on main.
Let me know if require anything else to setup
